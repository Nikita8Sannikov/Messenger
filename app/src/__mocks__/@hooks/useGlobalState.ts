const mockUseAppSelector = jest.fn();
const mockUseAppDispatch = jest.fn();

export const useAppSelector = mockUseAppSelector;
export const useAppDispatch = mockUseAppDispatch;

export { mockUseAppSelector, mockUseAppDispatch };
