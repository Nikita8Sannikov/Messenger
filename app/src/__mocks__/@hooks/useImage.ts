const useImage = (src: string | null): string => {
	return src || "/assets/images/noPhoto.jpg";
};

export default useImage; 