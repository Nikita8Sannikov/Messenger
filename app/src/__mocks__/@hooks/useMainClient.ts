const mockMainApi = {
	getAnotherUser: jest.fn(),
	signIn: jest.fn(),
	signUp: jest.fn(),
	logout: jest.fn(),
	uploadAvatarAuth: jest.fn(),
	getFriendsNotification: jest.fn(),
	getMessageNotification: jest.fn(),
	openFile: jest.fn(),
};

const useMainClient = jest.fn().mockReturnValue({ 
	mainApi: mockMainApi,
	getProfile: jest.fn(),
	existProfile: jest.fn(),
	lifeTimeExpire: jest.fn(),
	removeProfile: jest.fn(),
	downloadLogFile: jest.fn(),
});

export default useMainClient;
export { mockMainApi }; 