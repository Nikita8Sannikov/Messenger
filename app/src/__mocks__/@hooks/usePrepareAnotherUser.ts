export interface IUsePrepareAnotherUser {
  isLoading: boolean;
  isExistProfile: boolean;
}

const mockUsePrepareAnotherUser = (): IUsePrepareAnotherUser => ({
	isLoading: false,
	isExistProfile: true,
});

export default mockUsePrepareAnotherUser; 