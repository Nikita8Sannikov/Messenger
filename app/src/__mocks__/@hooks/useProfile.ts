export default () => ({
    on: jest.fn(),
    off: jest.fn(),
    user: {
        avatarUrl: jest.fn(),
        fullName: jest.fn()
    },
})