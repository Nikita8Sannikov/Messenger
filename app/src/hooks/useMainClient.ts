import { useContext } from "react";

import { MainClientContext } from "@components/main/MainClientProvider";

// Возврат сущности "MainClient"
export default function useMainClient() {
	return useContext(MainClientContext);
}
