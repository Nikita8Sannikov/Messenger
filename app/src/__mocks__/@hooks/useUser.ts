import { mockUserService } from "./useProfile";

const useUser = jest.fn().mockReturnValue(mockUserService);

export default useUser; 