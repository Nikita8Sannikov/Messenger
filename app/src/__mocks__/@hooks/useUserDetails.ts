import { mockUserDetailsService } from "./useProfile";

const useUserDetails = jest.fn().mockReturnValue(mockUserDetailsService);

export default useUserDetails; 