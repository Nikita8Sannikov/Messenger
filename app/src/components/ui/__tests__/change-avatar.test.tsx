import { ApiRoutes } from "common-types";
import { fireEvent, render, screen } from "@testing-library/react";

import ChangeAvatarComponent from "@components/ui/change-avatar";
import { mockMainApi } from "../../../__mocks__/@hooks/useMainClient";

describe("ChangeAvatarComponent", () => {
	const mockProps = {
		labelText: "Change Avatar",
		loading: false,
		onChange: jest.fn(),
		setLoading: jest.fn(),
	};

	beforeEach(() => {
		jest.clearAllMocks();
	});

	it("should render with label text", () => {
		render(<ChangeAvatarComponent {...mockProps} />);
        
		const label = screen.getByText(mockProps.labelText);
		expect(label).toBeInTheDocument();
	});

	it("should render input with correct id", () => {
		render(<ChangeAvatarComponent {...mockProps} />);
        
		const input = screen.getByTestId("input-image");
		expect(input).toHaveAttribute("id", `change-avatar-${mockProps.labelText}`);
	});

	it("should handle file upload with auth", () => {
		render(<ChangeAvatarComponent {...mockProps} />);
        
		const file = new File([ "test" ], "test.jpg", { type: "image/jpeg" });
		const input = screen.getByTestId("input-image");

		fireEvent.change(input, { target: { files: [ file ] } });
        
		const formData = new FormData();
		formData.append("avatar", file);
        
		expect(mockMainApi.uploadAvatarAuth).toHaveBeenCalledWith(
			ApiRoutes.uploadAvatar,
			formData,
			mockProps.setLoading,
			expect.any(Function),
		);
	});

	it("should not upload if no file is selected", () => {
		render(<ChangeAvatarComponent {...mockProps} />);
        
		const input = screen.getByTestId("input-image");
		fireEvent.change(input, { target: { files: [] } });

		expect(mockMainApi.uploadAvatarAuth).not.toHaveBeenCalled();
	});

	it("should show loading state", () => {
		render(<ChangeAvatarComponent {...mockProps} />);
        
		const button = screen.getByRole("button");
		const loadingWrapper = button.querySelector(".MuiButton-loadingWrapper");
		expect(loadingWrapper).toHaveStyle({ display: "contents" });
	});

	it("matches snapshot", () => {
		const { asFragment } = render(<ChangeAvatarComponent {...mockProps} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
