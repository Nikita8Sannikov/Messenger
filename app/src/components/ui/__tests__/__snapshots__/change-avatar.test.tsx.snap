// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeAvatarComponent matches snapshot 1`] = `
<DocumentFragment>
  <label
    class="input-label"
    for="change-avatar-Change Avatar"
  >
    <input
      accept="image/*"
      data-testid="input-image"
      hidden=""
      id="change-avatar-Change Avatar"
      type="file"
    />
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary input-label__button css-wyro15-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-loadingWrapper"
        style="display: contents;"
      />
      <p
        class="MuiTypography-root MuiTypography-caption input-label__button__text css-mpaqru-MuiTypography-root"
      >
        Change Avatar
      </p>
    </button>
  </label>
</DocumentFragment>
`;
